branch/check_status.py

```diff
@@ -41,6 +48,21 @@ def extract_issue_key(branch_to_check: str, valid_branch_names: str):
     return match.group(3)
 
 
+def extract_issue_key_pr_title(pr_title: str):
+    """
+    :param pr_title: The full PR title string from GitHub
+    :return: the extracted issue key
+    """
+    match = re.match(
+        r"^([a-zA-Z0-9]+-[0-9]+)\s\|\s\w.*$",
```

Comment:
Do we really also want to enforce the whitespace before the `|` here? Some of the PRs in `oasys-core` currently have the format `OSS-???| Title` and some `OSS-??? | Title`. Of course it would be more consistent to always use the whitespace, but I don't think CI should fail because of that (or does that simplify additional automation that uses the title?)

Reply:
> or does that simplify additional automation that uses the title?)

That is exactly the reason for the tight formatting rule, yeah. That makes dealing with the incoming title in the webhooks to JIRA automation possible.
I don't much care for with or without a preceding whitespace before the pipe character, it just needs to be always the same.